authoring-new-actions.md

```diff
@@ -0,0 +1,130 @@
+
+## ReadMe.md &amp; action.yml files
+
+Once the Action implementation is complete, ensure that it is documented well in the readme.md file at the root of the repo of the Master branch. Please note that this content is used when the Action is published to the GitHub Marketplace. Hence including a sample workflow which covers the best usage of the Action is always recommended.
+
+Also ensure that the Action.yml is present in the root of the repo and has the Action name &amp; description well defined. Also choose the icon &amp; color suitably under branding as these are used when the Action is listed in Marketplace catalog.
+
+Note: All actions published from **Azure** GitHub org are branded using the Azure icon irrespective of the icon used in the action.yml. This is to leverage the recall value of Azure brand and icon as compared to the individual icons of various Azure services. ![](RackMultipart20201207-4-o60y8a_html_71d21847106d5771.png)
+
+## Action Versioning
+
+Versioning guidance for GH action that should be adhered to before an action is released:
+
+URL: [https://github.com/actions/toolkit/blob/master/docs/action-versioning.md](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Factions%2Ftoolkit%2Fblob%2Fmaster%2Fdocs%2Faction-versioning.md&amp;data=02%7C01%7Cushan%40microsoft.com%7C8aae17deba8746e1a01c08d758875665%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637075210643253612&amp;sdata=%2BePOVjbQc4bH1FYUlbCFx%2BNCjbfXSGd%2F2P96ncGWaII%3D&amp;reserved=0)
+
+1. For a JS based action, don&#39;t check node\_modules into master: This is to avoid users to pick dependency on master branch which could be constantly churning. Hence node\_modules shouldn&#39;t be in master. Users are recommended to pick up only releases/v\* version.
+2. Create a release branch for each major version
+3. When ready for a stable release, add a major version tag (example V1)
+4. Create releases for minor and patch version updates (e.g. v1.2.3)
+5. Introduce a new major version of the branch (releases/v2) and tag (v2) if changes break the existing workflow.
+6. For pre-release/preview versions, ensure to name the tag to indicate the same. ex: v0-BETA. There is a system-defined pre-release tag you can leverage while publishing in the marketplace.
+
+![](RackMultipart20201207-4-o60y8a_html_259604dbe9d18c1d.png)
+
+## Publish the action to Marketplace
+
+### Pre-requisites:
+
+- Legal OSS approval process for the repo is complete. Search for the status of the work item created with your repo name here: https://ossmsft.visua lstudio.com/Reviews/\_workitems/ .
+
+&quot;DO NOT PUBLISH THE ACTION TO THE MARKETPLACE UNTIL LEGAL REVIEW IS COMPLETE&quot;
+
+- Squash merge all commits in Master branch and remove all previous git history from your development phase to ensure a clean state when the action is published
+- Versioning guidelines mentioned above section are adhered to and you have a version tag
```

Comment:
Ensure versioning guidelines mentioned above are adhered to

Reply:
modified the section.